src/core/public/saved_objects/saved_objects_client.ts

```diff
@@ -193,19 +233,24 @@ export class SavedObjectsClient {
    * @property {boolean} [options.overwrite=false]
    * @returns The result of the create operation containing created saved objects.
    */
-  public bulkCreate = (objects: BulkCreateOptions[] = [], options: KFetchQuery = {}) => {
+  public bulkCreate = (
+    objects: SavedObjectsBulkCreateObject[] = [],
+    options: SavedObjectsBulkCreateOptions = { overwrite: false }
+  ) => {
     const path = this.getPath(['_bulk_create']);
-    const query = pick(options, ['overwrite']) as Pick<KFetchQuery, 'overwrite'>;
+    const query = { overwrite: options.overwrite };
 
-    const request: ReturnType<SavedObjectsApi['bulkCreate']> = this.request({
+    const request: ReturnType<SavedObjectsApi['bulkCreate']> = this.savedObjectsFetch(path, {
       method: 'POST',
-      path,
       query,
-      body: objects,
+      body: JSON.stringify(objects),
```

Comment:
Optional: shouldn't HTTP.fetch internally call stringify? OTOH if an object cannot be stringified, it's better an exception raised in plugin code.

Reply:
stringifying inside `http.fetch` could be a nice convenience, but our current `http.fetch` behaviour makes this awkward. Becasue we throw on non-2xx responses, consumers of `http.fetch` will usually treat exceptions as "the server rejected our request" and probably won't correctly handle the true exceptions/errors such as a circular reference.

If an exception in `http.fetch` always meant something unexpected happened it would be safer to stringify the body. We could also use a discriminated union to type responses similar to https://github.com/infinitered/apisauce/blob/master/apisauce.d.ts#L36-L58. This would give you type hints that your code isn't handling error cases. When you reject/throw an error we loose all type information.

(On a side note, the non-2xx response throwing really tripped me up cause there's no type hints that warn you that it might happen, so my expectation was that the http.fetch API is the same as `window.fetch`)